### PR TITLE
Add integration-like to tests to cover full reconciliation loop

### PR DESCRIPTION
**Pull request by lwolf**

### CODE REVIEW

File: controllers/reconcile_fake_test.go [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+		namespace = "testns"
+	)
+	testCases := map[string]struct {
+		consumer              *konsumeratorv1alpha1.Consumer
+		expDeployAnnotation   map[string]map[string]string
+		expContainerResources map[string]corev1.ResourceRequirements
+		expContainerEnv       map[string]map[string]string
+	}{
+		"deployment should have minimum resources without metrics provider": {
+			consumer: &konsumeratorv1alpha1.Consumer{
```
Comment (hagen1778) [inline controllers/reconcile_fake_test.go]: looks like `ObjectMeta` and `Spec` are equal for all tests - let's move them to default struct

File: controllers/reconcile_fake_test.go [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func initReconciler(consumer *konsumeratorv1alpha1.Consumer) (client.Client, *controllers.ConsumerReconciler) {
+	var s = runtime.NewScheme()
+
+	_ = clientgoscheme.AddToScheme(s)
```
Comment (hagen1778) [inline controllers/reconcile_fake_test.go]: why omitting errors?

File: controllers/reconcile_fake_test.go [hunk 3]
```diff
@@ -0,0 +1,286 @@
+package controllers_test
```
Comment (hagen1778) [inline controllers/reconcile_fake_test.go]: why different pkg name?

File: controllers/reconcile_fake_test.go [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+)
+
+func initReconciler(consumer *konsumeratorv1alpha1.Consumer) (client.Client, *controllers.ConsumerReconciler) {
+	var s = runtime.NewScheme()
+
+	_ = clientgoscheme.AddToScheme(s)
+	_ = konsumeratorv1alpha1.AddToScheme(s)
+	_ = appsv1.AddToScheme(s)
+
+	objs := []runtime.Object{
```
Comment (hagen1778) [inline controllers/reconcile_fake_test.go]: no need to build a slice. 
```
cl := fake.NewFakeClientWithScheme(s, consumer)
```
should work

File: controllers/reconcile_fake_test.go (hunk 2 above)
Comment (lwolf) [reply controllers/reconcile_fake_test.go]: honestly, have no idea. all the docs, including examples in the source code omit errors during schema registration.

File: controllers/reconcile_fake_test.go (hunk 3 above)
Comment (lwolf) [reply controllers/reconcile_fake_test.go]: for more blackbox-like testing.

**Pull request merged**